### DISCUSSION TITLE
What to do with gateways like the KM200 MB LAN 2

### DISCUSSION THREAD

**Discussion by oliof** [emsesp/EMS-ESP32#2263] [Ideas]
### DESCRIPTION

As shortly discussed in discord, this is an issue to capture and document the ideas about what to do with official gateways such as the KM200 MB LAN 2. Some of them are listed as supported in the documentation; which means that ems-esp detects them and offers basic device information (bus id, device id, firmwaere version), but no gateway specific information is made available.

In a way, ems-esp provides a local superset of what's available via the official gateways and mobile apps. The mobile apps allow for limited control of boilers and thermostats (mostly schedules and set point information, and extra warm water production) and provide general energy usage stats (up to one year, in relatively low resolution),  as well as outside and room thermostat values if connected. Any and all of this can be done via EMS-ESP.

Two things that the gateway provides EMS-ESP doesn't are:

 1. maintenance access for installers/service technicians. In my experience, service technicians don't even know this exists or what to do with it.
 2. easy remote access to some controls mediated via the cloud connection of the gateway. Whether this is more or less secure than self-managed remote access to ems-esp depends on your threat scenarios and your capacity to provide secure remote access to your home network. 

Useful information about a gateway, if any were to be added, might be

 - online status (if it even knows)
 - connectivity log (i.e. what connected via the gateway when, useful for audit logs when you enabled the service technician function)
 - possibly status and audit information
 - _maybe_ firmware version reports, but since the gateways look for updates after power up and the manufacturers fully control the firmware and update mechanism, this would likely be a mostly static value (and we can already look it up in device information)
 

### REQUESTED INFORMATION

_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Searched the issue in [issues](https://github.com/emsesp/EMS-ESP32/issues)
- [X] Searched the issue in [discussions](https://github.com/emsesp/EMS-ESP32/discussions)
- [X] Searched the issue in the [docs](https://docs.emsesp.org/Troubleshooting/)
- [X] Searched the issue in the [chat](https://discord.gg/3J3GgnzpyT)
- [X] Provide the System information in the area below, taken from `http://<IP>/api/system`

```json
  {"system":{"version":"3.7.2-dev.0","uptime":"000+12:07:40.159","uptimeSec":43660,"platform":"ESP32","cpuType":"ESP32-D0WD","arduino":"Tasmota Arduino v2.0.17","sdk":"4.4.8.240628","freeMem":132,"maxAlloc":75,"freeCaps":82,"usedApp":1740,"freeApp":244,"partition":"app0","resetReason":"Software reset CPU / Software reset CPU","psram":false,"model":""},"network":{"network":"Ethernet","hostname":"ems-esp","TxPowerSetting":78,"staticIP":false,"lowBandwidth":false,"disableSleep":true,"enableMDNS":true,"enableCORS":false,"APProvisionMode":"disconnected","APSecurity":"wpa2","APSSID":"ems-esp"},"ntp":{"NTPStatus":"connected","enabled":true,"server":"time.google.com","tzLabel":"Europe/Amsterdam"},"mqtt":{"MQTTStatus":"connected","MQTTPublishes":9772,"MQTTQueued":0,"MQTTPublishFails":0,"MQTTReconnects":1,"enabled":true,"clientID":"ems-esp","keepAlive":60,"cleanSession":false,"entityFormat":0,"base":"ems-esp","discoveryPrefix":"homeassistant","discoveryType":0,"nestedFormat":1,"haEnabled":true,"mqttQos":0,"mqttRetain":false,"publishTimeHeartbeat":60,"publishTimeBoiler":10,"publishTimeThermostat":10,"publishTimeSolar":10,"publishTimeMixer":10,"publishTimeWater":10,"publishTimeOther":10,"publishTimeSensor":10,"publishSingle":false,"publish2command":false,"sendResponse":true},"syslog":{"enabled":false},"sensor":{"temperatureSensors":0,"temperatureSensorReads":0,"temperatureSensorFails":0,"analogSensors":0,"analogSensorReads":0,"analogSensorFails":0},"api":{"APICalls":0,"APIFails":0},"bus":{"busStatus":"connected","busProtocol":"Buderus","busTelegramsReceived":111729,"busReads":15418,"busWrites":3,"busIncompleteTelegrams":10,"busReadsFailed":0,"busWritesFailed":0,"busRxLineQuality":100,"busTxLineQuality":100},"settings":{"boardProfile":"E32","locale":"en","txMode":1,"emsBusID":11,"showerTimer":false,"showerMinDuration":180,"showerAlert":false,"hideLed":true,"noTokenApi":false,"readonlyMode":false,"fahrenheit":false,"dallasParasite":false,"boolFormat":1,"boolDashboard":1,"enumFormat":1,"analogEnabled":true,"telnetEnabled":true,"maxWebLogBuffer":25,"webLogBuffer":25,"modbusEnabled":false,"forceHeatingOff":false,"developerMode":false},"devices":[{"type":"boiler","name":"CBS/GB*72, Trendline, Cerapur, Greenstar Si","deviceID":"0x08","productID":123,"brand":"Buderus","version":"06.08","entities":80,"handlersReceived":"0x10 0x11 0x15 0x1C 0x18 0x19 0x34 0x2A 0x04","handlersFetched":"0x14 0x16 0x33 0x26 0x28","handlersPending":"0xBF 0xC2 0x1A 0x35 0xD1 0xE3 0xE4 0xE5 0xE6 0xE9 0xEA 0x2E 0x3B","handlersIgnored":"0x23"},{"type":"thermostat","name":"RC300/RC3*0, Moduline 3000/1010H, CW400, Sense II, HPC410","deviceID":"0x10","productID":158,"brand":"","version":"11.09","entities":56,"handlersReceived":"0x06 0xA2 0x02BA 0x02BB 0x02BC 0x02BD 0x02BE 0x02BF 0x02C0 0x031D 0x0267","handlersFetched":"0x02A5 0x02B9 0x02AF 0x029B 0x0291 0x0292 0x0293 0x0294 0x02F5 0x02F6 0x031B 0x023A 0x0240 0x0269","handlersPending":"0xA3 0x12 0x13 0x0471 0x02A6 0x02B0 0x029C 0x0472 0x02A7 0x02B1 0x029D 0x0473 0x02A8 0x02B2 0x029E 0x0474 0x02A9 0x02B3 0x029F 0x0475 0x02AA 0x02B4 0x02A0 0x0476 0x02AB 0x02B5 0x02A1 0x0477 0x02AC 0x02B6 0x02A2 0x0478 0x02CC 0x0467 0x02CE 0x0468 0x02D0 0x0469 0x02D2 0x046A 0x031E 0xBB 0x023E","handlersIgnored":"0xBF"},{"type":"gateway","name":"KM200, MB LAN 2","deviceID":"0x48","productID":189,"brand":"","version":"04.08","entities":0,"handlersIgnored":"0xF9 0xEF 0xF7 0xF6 0x04 0x0CBD 0x0CBC 0x03 0x0269 0x026A 0x026B 0x026C 0x026D 0x023A 0x06 0x02B9 0x02BA 0x02BB 0x02BC 0x18 0x19 0x1C 0x25 0x33 0xBF 0xC0 0xC1 0x0240 0x0291 0x0292 0x0293 0x0294 0x029B 0x029C 0x029D 0x029E 0x02A5 0x02AF 0x02B0 0x02B1 0x02B2 0x02C3 0x02F5 0x02F6 0x02FF 0x0309 0x031B 0x0449"},{"type":"controller","name":"BC25","deviceID":"0x09","productID":125,"brand":"","version":"03.03","entities":0}]}

```

### TO REPRODUCE

_Steps to reproduce the behavior:_ N/A

### EXPECTED BEHAVIOUR
N/A

### SCREENSHOTS

N/A

### ADDITIONAL CONTEXT

As requested, attached  is a watch log of the KM200 MB LAN2 on my system, including the traffic after powering down and up again the gateway.

[km200-watch.log](https://github.com/user-attachments/files/17964840/km200-watch.log)

**Comment by MichaelDvP** [comment]
> maintenance access for installers/service technicians. In my experience, service technicians don't even know this exists or what to do with it.

If you know how to use this, please make a log while activating/deactivating all function of maintenance menu.

I can't see any data send from the gateway, only verision and this:
```
gateway(0x48) -> all(0x00), B, ?(0x0CBD), data: 0F
gateway(0x48) -> all(0x00), B, ?(0x0CBC), data: 30 30 36 30 33 34 31 39 30 46 41 33
gateway(0x48) -> all(0x00), B, ?(0x0CBC), data: 50 54 4A 6D 4E 75 50 65 53 68 70 38 37 79 6D 41 39 31 37 32 31 30 30 31 39 (offset 12)
```
The 0x0CBC looks like ASCII `006034190FA3PTJmNuPeShp879mA917210019`, maybe the online access key?
(there are also 0xF7/F6 telegrams about these telegrams)
All other telegrams are requests to devices for values. Some unknown telegrams, but we don't know what these are for. 
0xC0, 0xC1 looks like a thermostat error log.

@proddy 
The new pretty-telegram looks nice, but before it was much better to use. Searching for writes from gateway was a simple search for `gateway(0x48) -W-`, now it's impossible and i have to scroll all the log.

**Reply by proddy** [reply]
> The new pretty-telegram looks nice, but before it was much better to use. Searching for writes from gateway was a simple search for gateway(0x48) -W-, now it's impossible and i have to scroll all the log.

I changed it because the -> was to only show the direction of traffic and the W/R was confusing. But I see your point. Happy to revert it if you want. include it in your next PR

**Reply by MichaelDvP** [reply]
We don't need the arrow, we can just log `<src> R <<dst>`, that allows to search in text editor for `<src> W` or `W <dst>` to check for write of changes.

**Reply by tp1de** [reply]
My km200 does not send telegram type 0x0CBC. 
Since all API communication is encrypted with rijndael EBC .
This encryption is used by LAN API polling and as well while communicating with Cloud. 
The content might be the encryption key.

**Comment by tp1de** [comment]
The km200 gateway provides similar information as being available within end-user interface on the thermostat.
It secures that this data is available on the Bosch Cloud Services.  The Buderus / Bosch Apps give some basic interface to change temps, switchTimes and holidayModes. Alarms on the heating systems are send as well.

The km200 can be polled by local LAN as well using the API. (implemented with my ioBroker adapter or my Home Assistant Node-red flow)
These API messages are encrypted and reading/writing needs decrypytion / encryption.
The polling response time is very slow compared to ems-esp. 

The km200 has additional entities /information available which is not within ems-esp (yet):
- switchTimes
- holidayModes
- recordings for energy consumption (last 13 months) and temperatures
- system energy monitoring (tank levels for oil boilers etc)
